src/coreclr/src/System.Private.CoreLib/src/System/Reflection/RuntimeConstructorInfo.cs

```diff
@@ -280,6 +286,14 @@ internal void ThrowNoInvokeException()
             // check basic method consistency. This call will throw if there are problems in the target/method relationship
             CheckConsistency(obj);
 
+            if ((invocationFlags & INVOCATION_FLAGS.INVOCATION_FLAGS_RUN_CLASS_CONSTRUCTOR) != 0)
+            {
+                // Run the class constructor through the class constructor mechanism instead of the Invoke path.
+                // This avoids allowing mutation of readonly static fields, and initializes the type correctly.
+                RuntimeHelpers.RunClassConstructor(DeclaringType!.TypeHandle);
```

Comment:
Can we get here for module constructors? I believe `DeclaringType` is going to be null for them.

Reply:
Sigh, we can. I'll update the code to handle them by running the module constructor api.